docs/resolving_merge_conflicts.md

```diff
@@ -1,26 +1,46 @@
 ## A guide to resolving merge conflicts
 
-When you come to merge your pull request, you may find that conflicts prevent automated merging back into the master.  In some cases, GitHub supports resolution of these through its web interface.  However, probably due to file size, this is not currently supported for cl-edit.owl.
+When you come to merge your pull request, you may find that conflicts prevent automated merging back into the master.  In some cases, GitHub supports resolution of these through its web interface. However, probably due to file size, this is not currently supported for cl-edit.owl.
 
-The majority of the time, the conflict is trivial - due to addition of new terms to the same point in the file. Becuase terms are ordered in the file by ID, this happens whenever two edits add terms without any intervening IDs.  Trivial clashes are easy to spot - they involve whole term stanzas + declarations.  No content is deleted (although this   
+The majority of the time, the conflict is trivial - due to addition of new terms to the same point in the file. Because terms are ordered in the file by ID, this happens whenever two edits add terms without any intervening IDs. Trivial clashes are easy to spot - they involve whole term stanzas + declarations.  
 
-Occassionally non-trivial clashes will happen when two pull requests include edits to the same term or even the same axiom.  Ask an editor for help if you don't feel confident resolving these.  
+Occassionally non-trivial clashes will happen when two pull requests include edits to the same term or even the same axiom. Ask an editor for help if you don't feel confident resolving these.  
 
 ### SOP.
 
-1. Reserialise the Master file using ROBOT. This will fix any odd orderings, that may have been introduced during previous merges.
-    ```sh
-    sh ./run.sh ROBOT convert --input cl-edit.owl -f ofn --output cl-edit.owl
-   ```
+1. Reserialise the Master file using the ODK commands normalize_src or normalize_obo_src. These commands require installation of ROBOT. If not already installed, follow [the instructions here](http://robot.obolibrary.org).
```

Comment:
I think this should be "requires ODK". The commands are in the ODK container not in ROBOT.

Reply:
ODK deals with docker, right? I was able to run the commands without docker, like how I run the make normalise_xsd_string command.
When I initially tried this, 'make normalize_scr' would not run because my ROBOT path was not set up correctly. It gave an error that 'robot' was missing. Once that was fixed, the command ran. So it seems the minimum requirement might be to have ROBOT or ODK?

Reply:
normalise_xsd_string is a custom command for CL (see cl.Makefile):
```
normalise_xsd_string: $(SRC)
	sed -i.bak -E "s/Annotation[(](oboInOwl[:]hasDbXref [\"][^\"]*[\"])[)]/Annotation(\1^^xsd:string)/g" $<
	rm $<.bak
```

For noramlise_src it is ODK, but you are running it through Makefile 
Makefile is deployed through ODK into repos, however, noramlise_src requires ROBOT
```
.PHONY: normalize_src
normalize_src: $(SRC)
	$(ROBOT) convert -i $< -f ofn -o $(TMPDIR)/normalise && mv $(TMPDIR)/normalise $<
```
ROBOT is in docker odk, so if you run `sh run.sh` (which calls on docker, you can see this in the run.sh file) instead of `make` it should be there already (makes it easier for people). 

Does that make sense? took me forever to understand this whole concept, and im not even 100% sure i fully understand it lol

Reply:
Are these the pre-req steps to run these commands? https://oboacademy.github.io/obook/howto/odk-setup/

I try to stay away from docker when possible due to memory issues. If there is a way to limit these steps to the line command, we should include it as an alternate option at least.

When I open docker and run: sh ./run.sh ROBOT convert --input cl-edit.owl -f ofn --output cl-edit.owl, I get this message:

docker: Error response from daemon: OCI runtime create failed: container_linux.go:380: starting container process caused: exec: "ROBOT": executable file not found in $PATH: unknown.

I might not have the wrapper script set up properly and will check on that.

This illustrates the situation where there seem to be multiple prerequisites before following the instructions as written. Is the odk-setup link above the best place to start?

Reply:
> docker: Error response from daemon: OCI runtime create failed: container_linux.go:380: starting container process caused: exec: "ROBOT": executable file not found in $PATH: unknown.

ROBOT should not be caps > `sh ./run.sh robot convert --input cl-edit.owl -f ofn --output cl-edit.owl`

Reply:
Running 

sh ./run.sh robot convert --input cl-edit.owl -f ofn --output cl-edit.owl

did not give an error (robot in lower case). Thanks! I don't know how to confirm it works for solving merge conflicts without an example, but at least there was no error.

Of note, I have to run make normalise_xsd_string to resolve the xsd issue after running normalize_src.

Reply:
huh, I guess saving it as is using ROBOT strips the xsd:string stuff, wonder if this is the case in other ontologies - might need to consider doing a normalise step automatically too - let me go take a look

Reply:
tried of bspo which uses owl edit file (never noticed how most of the other ontologies i work with use obo edit files) and it seems fine, guess its a CL thing hence the custom cmd in CL - for this document you can add to do that given it is CL specific

Reply:
Thank you for checking!
So it's clear, is running...

sh ./run.sh robot convert --input cl-edit.owl -f ofn --output cl-edit.owl

... all that is needed, or does the editor have to run the normalize_(obo)_src command in addition to that?

Reply:
running `sh run.sh noramlize_src` (or `sh run.sh normalize_obo_src`) is all that is neeeded -> that command basically runs `robot convert --input cl-edit.owl -f ofn --output cl-edit.owl`

For CL, just note that you have to run normalise_xsd_string after I guess

Reply:
Thanks, @shawntanzk. For setting up docker, It looks like odk.sh needs to be in the src/ontology/ directory for this to work (per the instructions here: https://oboacademy.github.io/obook/howto/odk-setup/).

Is that correct? I'm surprised it already isn't there if that is the case.

Reply:
`sh run.sh normalize_src`

`docker: Error response from daemon: OCI runtime create failed: container_linux.go:380: starting container process caused: exec: "normalize_src": executable file not found in $PATH: unknown.`

Reply:
`sh run.sh make normalize_src`